packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -120,11 +117,17 @@ export function TransferPanel() {
 
   const [transferring, setTransferring] = useState(false)
 
-  const [l1Amount, setL1AmountState] = useState<string>('')
-  const [l2Amount, setL2AmountState] = useState<string>('')
-
   const isSwitchingL2Chain = useIsSwitchingL2Chain()
 
+  // Link the amount state directly to the amount in query params -  no need of useState
+  // Both `amount` getter and setter will internally be useing useArbQueryParams functions
+  const [{ amount: queryAmount }, setQueryParams] = useArbQueryParams()
+  const amount = queryAmount && !isNaN(+queryAmount) ? queryAmount : '0'
+  const amountNum = parseFloat(amount) // just a numerical variant of amount which passes down in input component
+  const setAmount = (newAmount: string) => {
+    setQueryParams({ amount: newAmount })
+  }
```

Comment:
Maybe we should wrap this into `useCallback`, to be safe:

```suggestion
  const setAmount = useCallback((newAmount: string) => {
    setQueryParams({ amount: newAmount })
  }, [setQueryParams])
```

Reply:
Makes sense. Done.